### ISSUE TITLE
Live.me requires authentication now - Anonymous access no longer supported!

### ISSUE THREAD

**Issue by thecoder75** [thecoder75/liveme-api#15]
Getting the replays now requires additional parameters related to authentication of having a Live.me account now.  

Need help reversing the calls and documenting authentication procedures and stuff in order to add/update the module to fix the issue.

Originally received notice in [this issue](https://github.com/thecoder75/liveme-pro-tools/issues/23) on the LiveMe Pro Tools repository.

**Comment by thegeezer9999** [comment]
I remember them doing something like this before, and someone on Discord worked out that it expected a string of a certain type, as long as this was included it accepted the call. Unfortunately I cannot remember who it was, but maybe post this request on Discord ? I personally have no issue with logging in.

**Comment by Brapsfemy** [comment]
Not sure how to implement this for logins via social media.
There's a white list for the redirect URLs when logging in with various social media. In order to add the functionality to login with those, a new application would need to be registered for each.

As far as the standard login flow, it is as follows:
```
Step 1:
	GET:  https://live.ksmobile.net/channel/signin
	Data: {
		name: Email
		password: base64( password ),
		sr: 1 
	}
	Response: {
		status: Status code,
		msg: Any misc info
		data: {
			access_token: Access token
		}
	}
Step 2:
	GET:  https://live.ksmobile.net/channel/login
	Data: {
		thirdchannel: 6 ( Not sure what this is for ),
		androidid: Randomly generated UUID. For info on how it's generated, see below,
		countrycode:  ,
		reg_type: 108,
		access_token: First 21 characters of access token from step 1
	}
	Response: {
		status: Status code,
		msg: Any misc info,
		data:	{
			user: {
				uid: The user's unique ID,
				nickname: User's username,
				face: Profile picture I believe,
				level: User's level,
				reward: Not sure what this is,
				country: User's country code,
				area: Related to users country,
				short_id: User's short ID,
				diamond_left: Number of diamonds the user has,
				reg_time: I assume the token regen time in EPOC
			}
		}
	}
```

**Comment by Brapsfemy** [comment]
Even while passing the correct data in a get request, the server responds with:
```
{"status":"400","msg":"tuid,token,androidid can't empty","data":[]}
```

**Comment by anadee** [comment]
I'm trying to do my own thing in Java and I played around with this a bit and got it working with an email login.
The header "d" is required but seems like it can be any non empty string
The 'thirdchannel' parameter needs to be 6

Also androidid, countrycode, reg_type doesn't seem to be required for https://live.ksmobile.net/channel/login request (Step 2) 


```
GET http://live.ksmobile.net/live/getreplayvideos
Headers d:1

Data: {
    userid: user id of the replay owner
    page_index: 1
    page_size: 10
    thirdchannel: 6
    token: data.token (from Step 2, this was missing from Brapsfemy's example response)
    tuid: your user id (data.user.uid from Step 2)
}

Response: {
    {
	"status": "200",
	"msg": "",
    "recordSaveUserType": "1"
	"data": {
        "time": "1524093015",
		"video_info": [
            ...
        ]
    }
}
```

**Comment by thecoder75** [comment]
Thank you everyone, I'll update the API module by the weekend with the new commands.

**Comment by QuickFix998** [comment]
For anyone looking for a quick fix and is able to build liveme-pro-tools on their own, follow Brapsfemy and  anadee's method to get your token# and tuid#

Download the source, do your yarn install to get all the node_modules

Once they are all downloaded, go to the modules folder, open liveme-api and edit index.js
Change line 37 to axios.defaults.headers.common['d'] = '1';
Change line 124 to return httpGet(`${LM_GETREPLAYVIDEOS}?userid=${uid}&page_size=${count}&page_index=${page}&thirdchannel=6&token=yourtoken&tuid=yourtuid`);

I have no coding experience so I apologize if it is hard to follow, but it is working for me

**Comment by alansage** [comment]
I'm missing something here, in step 1 all I get is 

{"status":"400","msg":"param error","data":[]}

Guessing I'm malforming my curl request, I'm not that hot on curl so any clues welcome.  (Trying something outside of liveme-pro-tools)

**Comment by thecoder75** [comment]
New commands have been added to this module.  All that is left is adding the new user interfaces and handlers to the various tools that rely on this module.
